### PR TITLE
Add extension for Sequel transactions

### PR DESCRIPTION
**Pull request by waiting-for-dev**
## Overview

We add a `Dry::Operation::Extensions::Sequel` module that, when included, gives access to a `#transaction` method. This method wraps the yielded steps in a [Sequel](https://sequel.jeremyevans.net/) transaction, rolling back in case one of them returns a failure.

The extension expects the including class to define a `#db` method giving access to the Sequel database definition:

```ruby
class MyOperation < Dry::Operation
  include Dry::Operation::Extensions::Sequel

  attr_reader :db

  def initialize(db:)
    @db = db
  end

  def call(input)
    attrs = step validate(input)
    user = transaction do
      new_user = step persist(attrs)
      step assign_initial_role(new_user)
      new_user
    end
    step notify(user)
    user
  end

  # ...
end
```

Default options for the `#transaction` options (which delegates to Sequel [transaction method](https://sequel.jeremyevans.net/rdoc/files/doc/transactions_rdoc.html)) can be given both at include time with `include Dry::Operation::Extensions::Sequel[isolation: :serializable]`, and at runtime with `#transaction(isolation: :serializable)`.

As with the ActiveRecord extension, savepoints on nested transactions are not supported yet

### CODE REVIEW

**Comment by waiting-for-dev** [comment]
Waiting for https://github.com/dry-rb/dry-operation/pull/22

**Comment by waiting-for-dev** [comment]
Setting back to draft as we'll need to apply changes as in https://github.com/dry-rb/dry-operation/pull/24

**Comment by waiting-for-dev** [comment]
Applying changes as in https://github.com/dry-rb/dry-operation/pull/25

File: spec/integration/extensions/active_record_spec.rb [hunk 1]
```diff
@@ -156,7 +156,7 @@ def initialize(model)
       def call
         transaction do
           step create_record
-          transaction(requires_new: true) do
```
Comment (timriley) [inline spec/integration/extensions/active_record_spec.rb:159]: Was this change intended to be part of this PR? It's for AR, not Sequel.

File: spec/integration/extensions/sequel_spec.rb [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+        Success(db[:users].insert(name: "John"))
+      end
+    end.new(db: db)
+
+    expect(db).to receive(:transaction).with(isolation: :serializable)
+
+    instance.()
+  end
+
+  xit "works with `requires_new` for nested transactions" do
```
Comment (timriley) [inline spec/integration/extensions/sequel_spec.rb:118]: Is `requires_new` also a Sequel thing? I recall seeing this skipped test from the AR extension... is it a copy/paste mistake?

**Review by timriley — APPROVED** [review]
Couple of minor things to check, but otherwise looks good!

File: spec/integration/extensions/sequel_spec.rb (hunk 2 above)
Comment (waiting-for-dev) [reply spec/integration/extensions/sequel_spec.rb]: Same... Sequel uses `savepoint`. Fixed. Thanks for spotting it.
